conjure-java-core/src/main/java/com/palantir/conjure/java/util/ErrorGenerationUtils.java

```diff
@@ -145,15 +150,27 @@ public static void addAllParametersWithSafetyAnnotationsToMethodBuilder(
 
     public static ParameterSpec buildParameterWithSafetyAnnotation(
             TypeMapper typeMapper, FieldDefinition argDefinition, boolean isSafe) {
-        return buildParameterWithSafetyAnnotationInternal(typeMapper, argDefinition, isSafe)
+        return buildParameterWithSafetyAnnotationInternal(typeMapper, argDefinition, isSafe, true)
                 .build();
     }
 
-    private static ParameterSpec.Builder buildParameterWithSafetyAnnotationInternal(
+    public static ParameterSpec buildUnsanitizedParameterWithSafetyAnnotationAndJsonProperty(
```

Comment:
nit: not necessarily asking for a change, but the long-name of this method, which seems to do a lot and is really hard to read, makes me think that we ought to rely more directly on the builder pattern here, and e.g. have _separate_ methods that do
- Build the parameter spec
- sanitize
- add safety annotation
- add json property

Then these can be combined in whichever way we need where need to.

For instance, maybe we could:
- Remove `WithSafetyAnnotation` from the method names (it seems to always be there?)
- Expose `buildParameterWithSafetyAnnotationInternal` as `buildParameter`
  - We could remove the existing `buildParameterWithSafetyAnnotation` which doesn't bring much by itself
- Add a `addJsonProperty` util method that takes the builder and arg definition